src/charm.py

```diff
@@ -196,11 +230,7 @@ def _config_file(self, event: EventBase) -> str:
         config.update(self._server_config())
         config.update(self._integrations_config())
         config.update(self._prometheus_config())
-
-        # Don't accidentally destroy the Loki config by passing it
-        # `None` or `PebbleReady` or a `RelationEvent`
-        if isinstance(event, (LokiPushApiEndpointJoined, LokiPushApiEndpointDeparted)):
-            config.update(self._loki_config(event))
```

Comment:
This worries me, since I don't think we have integration tests around this. `if self.model.relations["logging-provider"]: ...` below is probably safe enough, but an integration test (or even unit test which fires off multiple events) would be great.

Reply:
We definitely won't destroy the loki config because the method is now stateless. I can add some tests for this, but I'm not sure exactly what you are asking for. I'm sure our integration tests already fire some events.

Reply:
I agree that the original cause of the bug no longer exists and that the code is probably exercised already in integration tests anyway. 

My real line of thinking is "if the truthiness of `self.model.relations[key]:` changes, what happens?" And the answer to that is probably "a loki config is generated which doesn't do anything".

`integration/test_alert_rules` looks like it already covers this case, or, if it doesn't, it's very close. I certainly wouldn't expect a completely separate test for a null hypothesis with the entire code flow changed, more of "hey, if you don't think this is already covered as a side effect of something else, adding a couple of lines to break the relation at the end to fire some other relation event and make sure everything is still ok is maybe a good safety measure."

Totally up to you, though.